Contributing.md

```diff
@@ -14,40 +14,73 @@ Conduct](CODE_OF_CONDUCT.md).
 Rustfmt requires a nightly compiler. Replace all invocations of `cargo` by
 `cargo +nightly` if necessary.
 
-At all times the environment variables `CFG_RELEASE_CHANNEL` and `CFG_RELEASE` must
-be set to either `nightly` or `beta`. Both should be set to the same value. The
-project supports [cargo-make](https://github.com/sagiegurari/cargo-make) which
-automatically sets these variables for you:
+At all times the environment variables `CFG_RELEASE_CHANNEL` and `CFG_RELEASE` must be set. `CFG_RELEASE_CHANNEL` must be set to either `nightly` or `beta`, and `CFG_RELEASE` should be set to your corresponding Rust version. The
+project supports and recommends [cargo-make][cargo-make] when working on rustfmt, because `cargo-make` automatically sets these variables for you
 
+The available tasks are listed in `Makefile.toml`.
+
+You can alternatively use `cargo` directly, but you'll have to set the `CFG_RELEASE` and `CFG_RELEASE_CHANNEL` environment variables and also provide the corresponding features.
+
+For example:
+```sh
+export CFG_RELEASE=1.45.0-nightly
+export CFG_RELEASE=nightly
 ```
-cargo +nightly make test
+(Windows users can use `set` to specify the environment variable values)
+
+## Building
+To build with `cargo make`:
+
+```sh
+cargo make build
 ```
 
-The available tasks are listed in `Makefile.toml`.
+Or alternatively with `cargo` directly:
+```sh
+cargo build --all-features
+# or
+CFG_RELEASE_CHANNEL=nightly CFG_RELEASE=1.45.0-nightly cargo build --all-features
+```
 
 ## Test and file issues
 
 It would be really useful to have people use rustfmt on their projects and file
 issues where it does something you don't expect.
 
 
-## Create test cases
+## Running tests
 
-Having a strong test suite for a tool like this is essential. It is very easy
-to create regressions. Any tests you can add are very much appreciated.
+The tests can be executed with `cargo make` or directly with `cargo` provided the corresponding environment variables have been set.
 
-The tests can be run with `cargo make test`. This does a number of things:
+To run tests with `cargo make`:
+
+```sh
+cargo make test
+```
+
+Or alternatively with `cargo` directly:
+```sh
+cargo test --all-features
```

Comment:
same comment as above, or mention that the second command are if the env variables aren't set for the shell